plugins.go

```diff
@@ -0,0 +1,185 @@
+package skipper
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"plugin"
+	"strings"
+
+	"github.com/zalando/skipper/filters"
+	"github.com/zalando/skipper/routing"
+)
+
+func findAndLoadPlugins(o *Options) error {
```

Comment:
This function could be split up into 3-4 functions, because the loops are independent as far as I understand.
Another idea would be to have these functions as methods (function with receiver *Options).

    func (o *Options) walkPlugindir() error
    func (o *Options) loadFilterPlugins() error
    func (o *Options) loadPredicatePlugins() error
    func (o *Options) loadDataclientPlugins() error